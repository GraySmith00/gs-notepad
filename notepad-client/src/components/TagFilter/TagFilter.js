import React from 'react';

const TagFilter = ({ setTagFilter }) => {
  return (
    <form className="tag-filter" onChange={e => setTagFilter(e.target.value)}>
      <select>
        <option value="all">All</option>
        <option value="work">Work</option>
        <option value="personal">Personal</option>
        <option value="hobby">Hobby</option>
      </select>
    </form>
  );
};

export default TagFilter;
