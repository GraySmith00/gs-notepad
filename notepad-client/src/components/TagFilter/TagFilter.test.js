import React from 'react';
import { shallow } from 'enzyme';

import TagFilter from './TagFilter';

describe('TagFilter component', () => {
  let wrapper;
  let mockSetTagFilter;

  beforeEach(() => {
    mockSetTagFilter = jest.fn();
    wrapper = shallow(<TagFilter setTagFilter={mockSetTagFilter} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setTagFilter on change', () => {
    expect(wrapper.find('.tag-filter').length).toEqual(1);

    wrapper.simulate('change', { target: { value: 'work' } });

    expect(mockSetTagFilter).toHaveBeenCalled();
  });
});
