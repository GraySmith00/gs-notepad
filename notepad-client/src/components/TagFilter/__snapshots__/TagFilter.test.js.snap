// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagFilter component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagFilter
    setTagFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select>
        <option
          value="all"
        >
          All
        </option>
        <option
          value="work"
        >
          Work
        </option>
        <option
          value="personal"
        >
          Personal
        </option>
        <option
          value="hobby"
        >
          Hobby
        </option>
      </select>,
      "className": "tag-filter",
      "onChange": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="all"
          >
            All
          </option>,
          <option
            value="work"
          >
            Work
          </option>,
          <option
            value="personal"
          >
            Personal
          </option>,
          <option
            value="hobby"
          >
            Hobby
          </option>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All",
            "value": "all",
          },
          "ref": null,
          "rendered": "All",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Work",
            "value": "work",
          },
          "ref": null,
          "rendered": "Work",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Personal",
            "value": "personal",
          },
          "ref": null,
          "rendered": "Personal",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hobby",
            "value": "hobby",
          },
          "ref": null,
          "rendered": "Hobby",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select>
          <option
            value="all"
          >
            All
          </option>
          <option
            value="work"
          >
            Work
          </option>
          <option
            value="personal"
          >
            Personal
          </option>
          <option
            value="hobby"
          >
            Hobby
          </option>
        </select>,
        "className": "tag-filter",
        "onChange": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="all"
            >
              All
            </option>,
            <option
              value="work"
            >
              Work
            </option>,
            <option
              value="personal"
            >
              Personal
            </option>,
            <option
              value="hobby"
            >
              Hobby
            </option>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "All",
              "value": "all",
            },
            "ref": null,
            "rendered": "All",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Work",
              "value": "work",
            },
            "ref": null,
            "rendered": "Work",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Personal",
              "value": "personal",
            },
            "ref": null,
            "rendered": "Personal",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hobby",
              "value": "hobby",
            },
            "ref": null,
            "rendered": "Hobby",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
